diff_test.go

```diff
@@ -84,6 +84,102 @@ func TestUpdateUserDiff(t *testing.T) {
 	assert.Equal(t, expected, res)
 }
 
+// Test that checks functionality for add/delete
+func TestRoleDiff(t *testing.T) {
+	old := Config{
+		Roles: []*Role{
+			{
+				ID:       "dev",
+				Accounts: []string{"123456789012"},
+			},
+		},
+	}
+
+	new := Config{
+		Roles: []*Role{
+			{
+				ID:       "admin",
+				Accounts: []string{"123456789012"},
+			},
+		},
+	}
+
+	res, err := new.ChangesFrom(old)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := Changes{
+		DeleteRoles: []string{"dev"},
+		AddRoles:    []string{"admin"},
+	}
+
+	assert.Equal(t, expected, res)
+}
+
+// Test that checks, create, update, and delete functionality for Roles
+func TestUpdateRoleDiff(t *testing.T) {
+	old := Config{
+		Roles: []*Role{
+			{
+				ID:       "admin",
+				Accounts: []string{"123456789012", "123456789013"},
+				Rules: []Rule{
+					{
+						Policy: "allow",
+					},
+				},
+			},
+			{
+				ID:       "dev",
+				Accounts: []string{"123456789012", "123456789013"},
+			},
+			{
+				ID:       "dev2",
+				Accounts: []string{"123456789012", "123456789013"},
+			},
+		},
+	}
+
+	// update admin role: remove rule, and account
+	// delete dev2
+	// create new role for dev3
+	new := Config{
+		Roles: []*Role{
+			{
+				ID:       "admin",
+				Accounts: []string{"123456789013"},
+			},
+			{
+				ID:       "dev",
+				Accounts: []string{"123456789012", "123456789013"},
+			},
+			{
+				ID:       "dev3",
+				Accounts: []string{"123456789012", "123456789013"},
+			},
+		},
+	}
+
+	res, err := new.ChangesFrom(old)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected := Changes{
+		UpdateRoles: []UpdateRole{
+			{
+				ID:           "admin",
+				AlteredField: []string{"Rules", "Accounts"},
```

Comment:
@jordiup it could be a good idea to extend on `AlteredField` to specify the old and new values (so we can show a git-like diff in the CLI)

Reply:
Cheers Chris, I agree, and think it would be a good extension 👌